internal/transformers/runtimesyntax.go

```diff
@@ -145,7 +145,9 @@ func (tx *RuntimeSyntaxTransformer) isFirstDeclarationInScope(node *ast.Node) bo
 }
 
 func (tx *RuntimeSyntaxTransformer) isExportOfNamespace(node *ast.Node) bool {
-	return tx.currentNamespace != nil && (node.ModifierFlags()&ast.ModifierFlagsExport != 0 || node.Flags&ast.NodeFlagsNestedNamespace != 0)
+	return tx.currentNamespace != nil &&
+		(node.ModifierFlags()&ast.ModifierFlagsExport != 0 ||
+			ast.IsModuleDeclaration(node) && ast.IsModuleDeclaration(tx.parentNode) && tx.parentNode.AsModuleDeclaration().Body == node)
```

Comment:
Is this line necessary assuming we don't drop the synthetic `export` in typeeraser.go?

Reply:
not for semantics. I thought that the emit might be different if we don't transform `namespace X { Y { Z { } }` back into `namespace X.Y.Z { }`, but that would be type emit -- and it might not matter if we changed that.